### ISSUE TITLE
Pass visit_id, scan_number and instrument as part of visit metadata 

### ISSUE THREAD

**Issue by DiamondJoseph** [DiamondLightSource/blueapi#552]
We require knowing the instrument, scan_number and visit_id for passing information into the central processing stream.
This information may be recoverable from the attach_metadata decorator and must be preserved when handling for visit metadata is changed.

**Comment by callumforrester** [comment]
So you want this in the start document?

**Comment by DiamondJoseph** [comment]
Yes. I've made the relevant ticket in dodal since the metadata attach is there (issue 666 get)
https://github.com/DiamondLightSource/dodal/issues/666

@tpoliaw needs this for forwarding to central msgbus

**Comment by DiamondJoseph** [comment]
Arguably we should have

```json
{
        "name": "start",
        "doc": {
            "uid": "44ca4429-ff8c-4495-8432-058b87c7f24c",
            "scan_id": 755659,
            "data_session": "cm12345-6"
            "instrument": "i22"
        }
    }
```

From which we can recover everything, and instrument is the only "non-standard" key on the StartDocument. Just need to integrate with the hooks for ScanNumber and DataSession.

**Comment by stan-dot** [comment]
@jacobfilik what are your thoughts on this?

**Comment by DiamondJoseph** [comment]
I've linked to @tpoliaw as he was working on forwarding messages to the central RMQ instance.

**Comment by tpoliaw** [comment]
Scan number, visit and if possible scan file path (or paths?) would be a good start. Associated DAQ ticket is [here](https://jira.diamond.ac.uk/browse/ACQP-157)

**Comment by DiamondJoseph** [comment]
We won't know the .h5 or .nxs file paths at the time the StartDocument is emitted: and those files won't have been created yet, but we could have all of them at the time of the StopDocument being emitted at the end of the scan?

We can guess where they are going to be, but the files won't be on disc yet.

**Comment by tpoliaw** [comment]
I think being in the stop document would be fine. The processing won't be able to run before the scan is complete anyway.
